swerve100/src/main/java/org/team100/lib/trajectory/FancyTrajectory.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package org.team100.lib.trajectory;
+
+import java.util.List;
+
+import org.team100.lib.motion.drivetrain.SwerveDriveSubsystem;
+
+import com.team254.frc2022.planners.DriveMotionPlanner;
+import com.team254.lib.geometry.Pose2d;
+import com.team254.lib.geometry.Pose2dWithCurvature;
+import com.team254.lib.geometry.Rotation2d;
+import com.team254.lib.swerve.ChassisSpeeds;
+import com.team254.lib.trajectory.TimedView;
+import com.team254.lib.trajectory.Trajectory;
+import com.team254.lib.trajectory.TrajectoryIterator;
+import com.team254.lib.trajectory.timing.CentripetalAccelerationConstraint;
+import com.team254.lib.trajectory.timing.TimedState;
+import com.team254.lib.trajectory.timing.TimingConstraint;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+
+public class FancyTrajectory extends Command {
+  /** Creates a new FancyTrajectory. */
+
+  SwerveDriveSubsystem m_robotDrive;
+  DriveMotionPlanner mMotionPlanner;
+  private TrajectoryIterator<TimedState<Pose2dWithCurvature>, TimedState<Rotation2d>> mCurrentTrajectory;
+
+  public FancyTrajectory(SwerveDriveSubsystem robotDrive ) {
+    // Use addRequirements() here to declare subsystem dependencies.
+
+    m_robotDrive = robotDrive;
+    mMotionPlanner = new DriveMotionPlanner();
+
+    addRequirements(m_robotDrive);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+
+    final double kMaxVel = 1.0;
+    final double kMaxAccel = 1.0;
+    final double kMaxVoltage = 9.0;
+
+    List<Pose2d> waypoints = List.of(
+                new Pose2d(0, 0, Rotation2d.fromDegrees(-90)),
+                new Pose2d(5, 5, Rotation2d.fromDegrees(0)));
+        // while turning 180
+        List<Rotation2d> headings = List.of(
+                Rotation2d.fromDegrees(90),
+                Rotation2d.fromDegrees(0));
+        // these don't actually do anything.
+        List<TimingConstraint<Pose2dWithCurvature>> constraints = List.of(
+                new CentripetalAccelerationConstraint(60));
+
+        // note there are static constraints in here.
+        DriveMotionPlanner mMotionPlanner = new DriveMotionPlanner();
```

Comment:
The bug you noticed yesterday is here, the declaration DriveMotionPlanner mMotionPlanner is *hiding* the member field of the same name.  remove the type ("DriveMotionPlanner") and then the variable will reference the member field, which is what you want.